mesonbuild/modules/features/feature.py

```diff
@@ -0,0 +1,356 @@
+# Copyright (c) 2023, NumPy Developers.
+# All rights reserved.
+import re
+from typing import (
+    Dict, Set, Tuple, List, Callable, Optional,
+    Union, Any, Iterable, cast, TYPE_CHECKING
+)
+from dataclasses import dataclass, field
+from ...mesonlib import File, MesonException
+from ...interpreter.type_checking import NoneType
+from ...interpreterbase.decorators import (
+    noKwargs, noPosargs, KwargInfo, typed_kwargs, typed_pos_args,
+    ContainerTypeInfo
+)
+from .. import ModuleObject
+
+if TYPE_CHECKING:
+    from typing import TypedDict
+    from typing_extensions import NotRequired
+    from ...interpreterbase import TYPE_var, TYPE_kwargs
+    from ...compilers import Compiler
+    from .. import ModuleState
+
+@dataclass(unsafe_hash=True, order=True)
+class ConflictAttr:
+    """
+    Data class representing an feature attribute that may conflict
+    with other features attributes.
+
+    The reason behind this class to clear any possible conflicts with
+    compiler arguments when they joined together due gathering
+    the implied features or concatenate non-implied features.
+
+    Attributes:
+        val: The value of the feature attribute.
+        match: Regular expression pattern for matching conflicted values
+               (optional).
+        mfilter: Regular expression pattern for filtering these conflicted values
+               (optional).
+        mjoin: String used to join filtered values (optional)
+
+    """
+    val: str = field(hash=True, compare=True)
+    match: Union[re.Pattern, None] = field(
+        default=None, hash=False, compare=False
+    )
+    mfilter: Union[re.Pattern, None] = field(
+        default=None, hash=False, compare=False
+    )
+    mjoin: str = field(default='', hash=False, compare=False)
+
+    def copy(self) -> 'ConflictAttr':
+        return ConflictAttr(**self.__dict__)
+
+    def to_dict(self) -> Dict[str, str]:
+        ret: Dict[str, str] = {}
+        for attr in ('val', 'mjoin'):
+            ret[attr] = getattr(self, attr)
+        for attr in ('match', 'mfilter'):
+            val = getattr(self, attr)
+            if not val:
+                val = ''
+            else:
+                val = str(val)
+            ret[attr] = val
+        return ret
+
+class KwargConfilctAttr(KwargInfo):
```

Comment:
```suggestion
class KwargConflictAttr(KwargInfo):
```

(with suitable renames later in the file)